fe/fe-core/src/main/java/org/apache/doris/planner/external/ExternalHiveScanProvider.java

```diff
@@ -69,7 +69,13 @@ public TFileFormatType getTableFormatType() throws DdlException, MetaNotFoundExc
     }
 
     @Override
-    public TFileType getTableFileType() {
+    public TFileType getTableFileType() throws MetaNotFoundException {
+        String location = hmsTable.getRemoteTable().getSd().getLocation();
+        if (location != null && !location.isEmpty()) {
+            if (location.startsWith("s3a") || location.startsWith("s3n")) {
+                return TFileType.FILE_S3;
+            }
+        }
```

Comment:
better to use `if else` to check the location for s3 and hdfs strictly, and throw exception if meeting unknown file type.